### PR TITLE
Adding documentation for stale projects

### LINKED ISSUE #4416: Missing documentation about the stale project feature in projects.md
**How to categorize this issue?**

/area documentation
/kind enhancement
/priority 3

**What would you like to be added**:
I was expecting to find some info about when and why a project is declared as stale in https://github.com/gardener/gardener/blob/master/docs/usage/projects.md, however there is nothing about it. 

Instead, such information is available in https://github.com/gardener/gardener/blob/master/docs/concepts/controller-manager.md#stale-projects-reconciler but I was not expecting to find it there. 

It would be nice have a link from projects.md to have some info or at least to link the other documentation resource.

**Why is this needed**:
UX with the documentation.

### PR DESCRIPTION
**Pull request by krgostev**
**How to categorize this PR?**

/area documentation
/kind enhancement

**What this PR does / why we need it**:
Documentation regarding the stale projects is missing in https://github.com/gardener/gardener/blob/master/docs/usage/projects.md. It would be better to include some details there of better UX.
**Which issue(s) this PR fixes**:
Fixes #4416 
**Special notes for your reviewer**:

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Review by rfranzke — DISMISSED** [review]
/lgtm

File: docs/usage/projects.md [hunk 1]
```diff
@@ -89,3 +89,7 @@ All newly created projects will only bind the owner to the `uam` role.
 The owner can still grant the `uam` role to other members if desired.
 For projects created before Gardener v1.8 the Gardener Controller Manager will migrate all projects to also assign the `uam` role to all `admin` members (to not break existing use-cases). The corresponding migration logic is present in Gardener Controller Manager from v1.8 to v1.13.
 The project owner can gradually remove these roles if desired. 
+
+## Stale Projects
+
+When a project is not actively used for some period of time the project is marked as "stale". This is done by controller called "Stale Projects Reconciler". Once the project is marked as stale there is a time frame in which if not used it will be deleted by that controller. More detailed information can be found [here](https://github.com/gardener/gardener/blob/master/docs/concepts/controller-manager.md#stale-projects-reconciler).
```
Comment (ialidzhikov) [inline docs/usage/projects.md]: I would suggest to use a relative link for the `docs/concepts/controller-manager.md#stale-projects-reconciler` reference. In this way user browsing the docs for vX.Y.Z will be navigated to the reference in version vX.Y.Z (not to the master branch always as done currently).

**Review by ialidzhikov — CHANGES_REQUESTED** [review]

File: docs/usage/projects.md (hunk 1 above)
Comment (krgostev) [reply docs/usage/projects.md]: Agree 100%, I just saw it was done that way in another doc file, so I thought that was the concept.

**Review by ialidzhikov — APPROVED** [review]
/lgtm

**Pull request merged**